### DISCUSSION TITLE
[@types/diff-match-patch] version 1.0.33 breaks compatibility

### DISCUSSION THREAD

**Discussion by andybalaam** [DefinitelyTyped/DefinitelyTyped#66677] [Issues with a @types package]
In [@types/diff-match-patch 1.0.32](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f6277f747a7b141e4888bba5718660a00c4a0f2e/types/diff-match-patch/index.d.ts#L16) the types are exported with e.g. `export class diff_match_patch`.

In [@types/diff-match-patch 1.0.33](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/diff-match-patch/index.d.ts#L100) it is `export = diff_match_patch`.

This breaks code that used to import like: `import { diff_match_patch as DiffMatchPatch } from "diff-match-patch";` but now must change to using `import DiffMatchPatch from "diff-match-patch";`

This change was caused by https://github.com/DefinitelyTyped/DefinitelyTyped/commit/abce566b5028334de3a6e56aeaccaca6c1413457

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment]
Hi @andybalaam!

Sorry this is causing you trouble. I believe the changes I made in https://github.com/DefinitelyTyped/DefinitelyTyped/commit/abce566b5028334de3a6e56aeaccaca6c1413457 are correct because diff-match-patch is CJS. Before the types were declaring it as ESM when they shouldn't have been.

Regardless, your use-case should still work. I tried [reproducing the issue](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgE2AMzQfRAQxgYwAtMw8i4cBnOAEXTQFkzCAFZuAXzjSghDgBEqDAFpcBQiNISBAbgCwAKCX4IAO0rxk4OAF44agKYB3WvSYS2EgBQBKBYtUat9PSnAA6YVlzA11gHJkCABzagAjHCgAawCAGjgA-DwIqNj7JSA), but it seems to work for me. This likely means there's some TSConfig difference in our setups. Can you include your TSConfig and/or figure a way to reproduce it so that I can take a deeper look into the issue?

**Reply by andybalaam** [reply]
@Methuselah96 - thank you for the reply. Here is the PR experiencing the issue: https://github.com/matrix-org/matrix-react-sdk/pull/11390

The build error is: `Error: src/utils/MessageDiffUtils.tsx(149,84): error TS2749: 'DiffMatchPatch' refers to a value, but is being used as a type here. Did you mean 'typeof DiffMatchPatch'?`

Build log: https://github.com/matrix-org/matrix-react-sdk/actions/runs/6158822355/job/16712386423?pr=11390

**Reply by Methuselah96** [reply]
Thanks, that's helpful.

The quick fix is to do as the error message suggests and use `typeof DiffMatchPatch`. I think this may have to do with the class and namespace merging making it so that `DiffMatchPatch` can no longer be referred to as both a value and a type (like a bare class can be). I will have to look into it further to know for sure, but I suspect this might be a TypeScript limitation with no better solution.

**Reply by andybalaam** [reply]
Thank you for the quick and helpful responses!
